### PR TITLE
Refactored client, removed vendored gql libs

### PR DESCRIPTION
**Pull request by vanpelt**
https://wandb.atlassian.net/browse/WB-5747

Description
-----------

This removes a couple vendored packages and makes a custom `GQLClient` that gives us more control over our client.  An added bonus is removing `subprocess32` as it was only for python2.

Testing
-------

How was this PR tested?

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points. If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------

* Added a dependency on graphql-core>=2.3.0
* Removed dependency on subprocess32
* Removed urllib3 pin to avoid conflicts, if you see urllib3 related errors run `pip install --upgrade urllib3`
* Improved Public API HTTP error messages

------------- END RELEASE NOTES --------------------

### CODE REVIEW

File: wandb/old/summary.py
Comment (vanpelt) [inline wandb/old/summary.py:15]: All changes below this line are black formatting

File: wandb/apis/normalize.py
Comment (raubitsj) [inline wandb/apis/normalize.py:27]: Was this path not used? if it was, what is the replacement?

**Review by raubitsj — APPROVED** [review]
Awesome.
Only question I have is about RetryError, how it was used before and why it isnt needed now.

File: wandb/apis/normalize.py
Comment (raubitsj) [reply wandb/apis/normalize.py:27]: Looking closer at this, it looks like the old behavior did a blanket exception catch so if requests happens to throw an exception it would get rethrown here,  without this, there is a risk of an unhandled exception (though the only I could find look unlikely in our cases (too many redirects, etc)

Comment (vanpelt) [reply wandb/apis/normalize.py:27]: This path was used for generating better graphql errors when the server returned an "error" payload.  Now the GQLClient just raises a generic `Exception` with the first error in the payload as the message.  We may want to raise a special error class for that case, I'll chew on it a bit.

Comment (raubitsj) [reply wandb/apis/normalize.py:27]: If you want to merge it now and let things shake out, that might be good too, as long as we closely think about the implications in this path as it is the highest risk of this change.

**Review by raubitsj — DISMISSED** [review]
graphql-core-3 doesnt support python3.5 so this will break.

Lets hold this PR for a week or two and see if we can do a 0.12.0 in early/mid august where this can land.
or you can relax the requirement but I dont know what that impacts.

**Comment by dannygoldstein** [comment]
I defer to @raubitsj on this, but @vanpelt this is really cool to see

File: wandb/apis/normalize.py
Comment (vanpelt) [reply wandb/apis/normalize.py:27]: @raubitsj I did some more work here.  Specifically [this commit](https://github.com/wandb/client/pull/2419/commits/152f887293a2aa02f0023cd2b92133c255c08ffc) would be good to review.  It loosens the graphql-core requirement so we could release this on the 0.11.x branch if we want.

`RetryError` used to be raised by gql when it's own internal retry logic failed.  I've now made `execute` only raise `requests.RequestExeception` errors.  I also handled two cases that used to suck.

1. If we get error details from the server, the `requests.HTTPError` error that we raise will have a better message.  This is especially useful in the public api.  For example:

## Before

```
runs = api.runs("foo/bar", filters={"summary.foo": "bar"})
HTTPError: 400 Client Error: Bad Request for url: https://api.wandb.ai/graphql
```

## After

```
runs = api.runs("foo/bar", filters={"summary.foo": "bar"})
HTTPError: Error 1054: Unknown column 'summary' in 'where clause'
```

2. If the server doesn't return valid JSON we raise a `requests.HTTPError: 500` with a message "http://localhost:8080/graphql did not return valid JSON" to handle invalid proxies or shutdown local instances.

File: wandb/apis/public.py
Comment (raubitsj) [inline wandb/apis/public.py:1692]: This looks like you wanted this to be a tuple but it isnt.
did you want:
```
retryable_exceptions=(requests.RequestException,),
```

digging into where it is used, im not sure it matters, but all other users seem to use a tuple here.

Comment (vanpelt) [reply wandb/apis/public.py]: Good catch, it still worked because we just do `except self._retryable_exceptions as e` which will work with a single value or a tuple, but best to be consistent.  Just fixed.

**Review by raubitsj — APPROVED** [review]
Lets do it!

**Pull request merged**
